[feat] implement users profile and wish/have perfumes

wanip17 opened the pull request:
## 👩🏻‍💻 작업 내용
- 위시향수/가진향수 불러오기/추가/삭제 구현
- 사용자 프로필을 불러오기/수정하기 구현

<!-- ## 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|기능이름|스크린샷 첨부| -->

## 🔍 참고 사항
- base: feat/dev-oauth
-  아직 login의 withdraw가 구현 안된 상태

## 🔘 관련 이슈
- close #

wanip17 committed:
feat: 유저 위시향수/가진향수 및 프로필 정보 구현

yaongmeow commented on src/main/java/com/ppu/ppu/user/User.java:
```diff
@@ -21,7 +22,7 @@ public class User {
     private UUID id;
 
-    @Column(name = "password", length = 255)
+    @Column(name = "password", length = 255, nullable = false)
```
password nullable false여도 오류 안나나요?? kakao login 쪽이 password가 없을텐데..!!

wanip17 replied:
오잉 그러네요! 제가 실행할 때는 문제가 없었는데, 다음 이유로 우연히? 문제가 없던 것 같습니다.
- dev-users 브랜치 에서는 null password에 대해서 bcrypt 실행 -> 일반적인 값이 들어감
- dev-withdraw에서는 리팩토링 과정에서 null 값이 그대로 저장
- But, dev-withdraw에서 기존 table을 그대로 사용해서 nullable=false가 그대로 통과

nullable = false 삭제하고 푸시할게요

yaongmeow reviewed (commented):
users 하위 perfume과 profile을 나눈 것까지는 좋은데, perfume 하위는 너무 잘게 쪼개진 것 같아요..!! 거의 메서드가 2개 ~ 3개 정도인 것 같아서요 합치는 것이 어떨까요?

규모가 커지면 나누는 게 맞는데, 지금 규모에서는 너무 작게 나누어진 것 같아요. 혹시 이유가 있었다면 말씀해주셔도 좋습니다.

wanip17 commented:
> users 하위 perfume과 profile을 나눈 것까지는 좋은데, perfume 하위는 너무 잘게 쪼개진 것 같아요..!! 거의 메서드가 2개 ~ 3개 정도인 것 같아서요 합치는 것이 어떨까요?
> 
> 규모가 커지면 나누는 게 맞는데, 지금 규모에서는 너무 작게 나누어진 것 같아요. 혹시 이유가 있었다면 말씀해주셔도 좋습니다.

perfume은 저도 원래 한 디렉토리에 모두 넣을 생각이었는데, wish/have의 기능이 명확히 나뉘다 보니 나누게 되었어요..! 아마 공통되는 부분을 리팩터링하면 파일이 줄을거 같긴 한데... 저도 고민이네요

wanip17 committed:
fix: password can be nullable

wanip17 committed:
Resolve conflict by Merging branch 'feat/dev-users' into 'develop'

# Conflicts:
#	src/main/java/com/ppu/ppu/auth/controller/AuthController.java
#	src/main/java/com/ppu/ppu/auth/service/AuthService.java
#	src/main/java/com/ppu/ppu/auth/service/OAuthService.java
#	src/main/java/com/ppu/ppu/user/UserRepository.java
#	src/main/java/com/ppu/ppu/user/UserService.java
